bugs/45ea695f.txt

```diff
@@ -0,0 +1,148 @@
+=== Begin SRU Template ===
+[Impact]
+Some image creators would like to lock down authorized ssh keys for closed
+appliance type vms. This new cloud-config directive will skip sourcing
+ssh public keys from the metadata service.
+
+[Test Case]
+# Provide cloud metadata public-keys and ignore them via allow_public_ssh_keys: false
+cat > md <<EOF
+instance-id: mytest
+local-hostname: mytest
+public-keys:
+- ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDSL7uWGj8cgWyIOaspgKdVy0cKJ+UTjfv7jBOjG2H/GN8bJVXy72XAvnhM0dUM+CCs8FOf0YlPX+Frvz2hKInrmRhZVwRSL129PasD12MlI3l44u6IwS1o/W86Q+tkQYEljtqDOo0a+cOsaZkvUNzUyEXUwz/lmYa6G4hMKZH4NBj7nbAAF96wsMCoyNwbWryBnDYUr6wMbjRR1J9Pw7Xh7WRC73wy4Va2YuOgbD3V/5ZrFPLbWZW/7TFXVrql04QVbyei4aiFR5n//GvoqwQDNe58LmbzX/xvxyKJYdny2zXmdAhMxbrpFQsfpkJ9E/H5w0yOdSvnWbUoG5xNGoOB blah@cloudkeys
+EOF
+
+cat > lxd.yaml <<EOF
+#cloud-config
+lxd:
+  init:
+    storage_backend: dir
+ssh_import_id: [chad.smith]
+allow_public_ssh_keys: false
+EOF
+
+for SERIES in xenial bionic disco eoan; do
+   echo '=== BEGIN ' $SERIES;
+   ref=$SERIES-proposed;
+   lxc delete test-$SERIES --force 2> /dev/null;
+   lxc-proposed-snapshot -p -P $SERIES $ref | egrep 'Creating|cloud-init';
+   lxc init $ref test-$SERIES;
+   lxc config set test-$SERIES user.user-data - < lxd.yaml;
+   lxc config set test-$SERIES user.meta-data - < md;
+   lxc start test-$SERIES;
+   lxc exec test-$SERIES -- cloud-init status --wait --long;
+   lxc exec test-$SERIES grep wait /var/log/cloud-init.log;
+   lxc exec test-$SERIES grep Trace /var/log/cloud-init.log;
+   lxc exec test-$SERIES grep Skipping /var/log/cloud-init.log | grep ssh;
+   echo -n "Expect 0 cloud keys in authorized_keys: ";
+   lxc exec test-$SERIES -- grep -c cloud /home/ubuntu/.ssh/authorized_keys;
+   echo -n "Expect 1 cloud key in meta-data: ";
+   lxc exec test-$SERIES -- grep -c cloud /var/lib/cloud/seed/nocloud-net/meta-data;
+   lxc stop test-$SERIES;
+   lxc delete test-$SERIES;
+done
+
+[Regression Potential]
+Minimal adds a simple lxd command on initialization.
```

Comment:
This doesn't look like it's applicable here. Copy/paste error?

Reply:
+1 copy paste error. updated. we could have regression if no public keys were published in default case if allow_public_ssh_keys wasn't specified.